cmd/buildah/images.go

```diff
@@ -506,27 +539,37 @@ func formattedSize(size int64) string {
 	return fmt.Sprintf("%.3g %s", formattedSize, suffixes[count])
 }
 
-// reposToMap parses the specified repotags and returns a map with repositories
-// as keys and the corresponding arrays of tags as values.
-func imageReposToMap(repotags []string) map[string][]string {
-	// map format is repo -> tag
+// imageReposToMap parses the specified names list and returns a map with repositories
+// as keys and a slice of tags and digests as values.
+func imageReposToMap(names []string) map[string][]string {
+	// map format is repo -> []tagOrDigest
 	repos := make(map[string][]string)
-	for _, repo := range repotags {
-		var repository, tag string
-		if strings.Contains(repo, ":") {
-			li := strings.LastIndex(repo, ":")
-			repository = repo[0:li]
-			tag = repo[li+1:]
-		} else if len(repo) > 0 {
-			repository = repo
-			tag = "<none>"
-		} else {
+	for _, name := range names {
+		if name == "" {
 			logrus.Warnf("Found image with empty name")
+			continue
+		}
+		named, err := reference.ParseNormalizedNamed(name)
+		if err != nil {
+			logrus.Warnf("Error parsing name %q: %v", name, err)
+			continue
+		}
+		if named.Name() != name {
+			logrus.Debugf("Name %q wasn't already in its normalized form (%q).", name, named.Name())
+		}
+		var repository, tagOrDigest string
+		repository = reference.TrimNamed(named).Name()
+		if tagged, ok := named.(reference.Tagged); ok {
+			tagOrDigest = tagged.Tag()
+		}
+		if digested, ok := named.(reference.Digested); ok {
+			digest := digested.Digest()
+			tagOrDigest = digest.String()
 		}
-		repos[repository] = append(repos[repository], tag)
+		repos[repository] = append(repos[repository], tagOrDigest)
 	}
 	if len(repos) == 0 {
-		repos["<none>"] = []string{"<none>"}
```

Comment:
just curious why we're getting rid of the <none> for the dangling images?  I thought we'd some docs telling folks to look for that tag to determine what was dangling or not?

Reply:
This moves us from inserting `<none>` as a recognized-as-fake value here to having the tabular output logic decide whether or not to output `<none>` based on whether or not there's a value to output, and to having the JSON output format be `,omitempty` for tag and digest fields.  If the docs are about non-JSON `buildah images` output, the end-result should look the same.

Reply:
Thanks @nalind, the `buildah images` is what I was most concerned about.